nixos/modules/services/networking/wpa_supplicant.nix

```
@@ -94,7 +94,7 @@ let
         ++ optional (pskString != null) "psk=${pskString}"
         ++ optionals (opts.auth != null) (filter (x: x != "") (splitString "\n" opts.auth))
         ++ optional (opts.priority != null) "priority=${toString opts.priority}"
-        ++ optional (opts.extraConfig != "") opts.extraConfig;
+        ++ optionals (opts.extraConfig != "") (filter (x: x != "") (splitString "\n" opts.extraConfig));
```

nit: `opts.extraConfig != ""` is not necessary
```suggestion
        ++ filter (x: x != "") (splitString "\n" opts.extraConfig);
```

---

I'll update it.

---

Done.